connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -384,6 +417,40 @@ public void onCompletion(RecordMetadata recordMetadata, Exception e) {
         return true;
     }
 
+    // Due to transformations that may change the destination topic of a record (such as
+    // RegexRouter) topic creation can not be batched for multiple topics
+    private void maybeCreateTopic(String topic) {
+        if (!isTopicCreationEnabled || topicCache.contains(topic)) {
+            return;
+        }
+        log.info("The task will send records to topic '{}' for the first time. Checking "
+                + "whether topic exists", topic);
+        Map<String, TopicDescription> existing = admin.describeTopics(topic);
+        if (!existing.isEmpty()) {
+            log.info("Topic '{}' already exists.", topic);
+            topicCache.add(topic);
+            return;
+        }
+
+        log.info("Creating topic '{}'", topic);
+        NewTopicCreationGroup topicGroup = topicGroups.values().stream()
+                .filter(group -> group.matches(topic))
+                .findFirst()
+                .orElse(defaultTopicGroup);
+        log.debug("Topic '{}' matched topic creation group: {}", topic, topicGroup);
+        NewTopic newTopic = topicGroup.newTopic(topic);
+
+        if (admin.createTopic(newTopic)) {
+            topicCache.add(topic);
+            log.info("Created topic '{}'", newTopic);
```

Comment:
WDYT about including the name of the topic creation group that was used (it's only included in debug logs so far) and the actual topic settings used to create the topic? Something like:
```suggestion
            log.info("Created topic '{}' using creation group '{}' and topic settings {}", newTopic, topicGroup, topicGroup.topicSettings());
```
if the `NewTopicCreationGroup.toString()` method output the name only, or
```suggestion
            log.info("Created topic '{}' using creation group {}", newTopic, topicGroup);
```
if the `NewTopicCreationGroup.toString()` method output the name and topic settings.

Reply:
👍 went with option 2.